app_dart/lib/src/service/datastore.dart

```diff
@@ -151,6 +151,9 @@ class DatastoreService {
         updates: 0,
       );
     } else {
+      if (previousStatusUpdates.length > 1) {
+        throw 'GithubBuildStatusUpdate should have no more than one entries on repository ${slug.fullName}, pr ${pr.number}, head ${pr.head.sha}';
```

Comment:
We should throw an actual exception here - maybe a `StateError`?

Reply:
And with this, you can get rid of the assert below.

Reply:
Done.